### DISCUSSION TITLE
2022-05-11 Budget Subcomittee Meeting

### DISCUSSION THREAD

**Discussion by andy-blum** [ddev/ddev#3828] [DDEV Advisory Group]
Previous meeting: #3681 
Next meeting: #3800

## Agenda:
- Start recording
- Ask somebody to take notes on the attached doc
- Financial reports
- Test runner purchase plan
- Discussion: Strategies to develop a co-maintainer
- Other discussion

## May 11, 2022 | [DDEV Budget Group](https://www.google.com/calendar/event?eid=NWkxdnAxa2xtajM2NDRraDUzNXY5azIyYXAgcmFuZHlAcmFuZHlmYXkuY29t)

Attendees: Andrew Berry [Cary Gordon](mailto:cgordon@chillco.com)[Christopher Wells](mailto:chris@redfinsolutions.com)[Randy Fay](mailto:randy@randyfay.com)[Ofer Shaal](mailto:shaalofer@gmail.com)

Agenda:

1. **Start recording ✅**
2. **Ask somebody to take notes on the attached doc ✅**
3. **Financial reports**
    - P&L (Profit & Loss) for the Localdev Foundation. Jan 1-May1, 2022.
      - ~7600 in income,
      - ~240 for MacStadium/CircleCI,
      - ~35 Wifi on the plane
      - ~$150 - upstream contributions &amp; support
    - We now have an M1 Mac through MacStadium - they&#39;ve refunded the costs plus given us one free.
    - There are two M1&#39;s Randy has that are being TestBots now because Platform.SH got him a daily driver machine.
    - Financials reports are generated through Wave Accounting, which is being used by LocalDev. There&#39;s no cost associated with Wave.
    - ~$7300 &quot;in the bank.

4. **Test runner purchase plan**
	- [See Randy's comment below](https://github.com/drud/ddev/discussions/3828#discussioncomment-2725435)
    - TWO AMD and one ARM Mac Mini ~$2100
    - Windows side - try a mini PC 8GB/256 ~$500. Buy 4 more if testing of this one works.
      1. AB: What about buying one beefy machine and treating it as multiple? RF: Nested Virtualization causes issues with Docker, WSL2, etc.
      2. AB: Cloud providers don&#39;t want to do the overhead for Nested, but on our own machine, maybe this is fine? RF: Windows Server + Virtualization + Docker/WSL is really 3 layers.
      3. CG: The machines are around $329 on Amazon. Smaller more machines is probably better than the single machine.
      4. It would be nice to have a Windows Server solution or something to help automate test-runner setup. (Licensing considerations, though.)
      5. AB: Electricity costs covered for this? RF: For now I&#39;m not super worried about it. Would be cool to know the usage though. Something like a [Smart Home Energy Monitor](https://www.amazon.com/Emporia-Monitor-Circuit-Electricity-Metering/dp/B08CJGPHL9/ref=sr_1_8?keywords=power+consumption+meter&amp;qid=1652282933&amp;sr=8-8) may help us understand more than just consumption, but also usage of individual devices.
      6. Test runners are isolated on the network in terms of current config. With GitHub Actions I could change so that every PR has to be manually authorized. But BuildKite - not sure if that can be done or how.
      7. CW: Is this a rough estimate for yearly outlay, or is this more of a big push now to catch up? RF: Probably more of a big push, but if we do expand our test coverage or expand scope then there may be more. Scale is always the issue.
      8. CG: We don&#39;t know what&#39;s gonna happen in the future, for sure.
      9. OS: Check out CodeSpaces for test-running. RF: GitHub CodeSpaces is a direct competitor for GitPod; there&#39;s great support for GitPod but not a lot of testing for that environment. We don&#39;t have real solutions to that yet. There are two major permutations of DDEV that don&#39;t have testing right now. Docker inside WSL2, and GitPod. GitHub CodeSpaces would be a third permutation, but it&#39;ll certainly take someone&#39;s time to figure that out.
5. **Discussion: Strategies to develop a co-maintainer**
    1. RF: Maybe Platform could help us fill this role. Would be cool to have Docs professionals working on the documentation. So that&#39;s one possible area to look toward.
    2. Another possibility is we currently have money coming in. It&#39;s not a full time dev salary, but it could be a push toward a full-time job. Hard to find that person who comes on part-time but wants it to become full-time. People like us are already fully booked.
    1. CG: there&#39;s an element of serendipity here. If you put it out there and put it out there enough, someone may appear.
    2. How do we market this? Twitter, release notes, a message that comes up with DDEV.
    3. AB: something that could help this is instead of focusing on a role, focus on mini-projects. Or focus on support. Start small, rather than scaring off with &quot;hiring an open-source co-maintainer&quot; :)
    4. Gentleman (missed name?) from Typo3 community who is in Germany is really good for us, he&#39;s a logical choice to maybe help watch the test-runner.
    5. Hurdles here are the history and complex skillsets needed. Go is divorced from PHP a bit. And Randy has a lot of experience with Docker/containerization/Linux.
    6. We probably do want to spend the money each year, not have any left over.
    7. Making some inroads in the WordPress community, maybe some connections there.
6. **Other discussion**
    1. We need to consider marketing, but not a 300k/yr contract.
    2. CG: So you&#39;re going to go ahead and buy the stuff? RF: Yes, we&#39;ve got consensus here. We&#39;re not &quot;voting&quot; _per se_, we are an advisory board, not a decision-making board.

# Recording

Watch the [recording](https://www.dropbox.com/s/f07afytrlr69744/GMT20220511-150846_Recording_1760x900.mp4?dl=0)

# Reports
![image](https://user-images.githubusercontent.com/112444/167495068-2777928a-782d-4add-860e-ab47a9d49753.png)

[Localdev Foundation Profit and Loss 2022-05-01.pdf](https://github.com/drud/ddev/files/8655184/Localdev.Foundation.Profit.and.Loss.2022-05-01.pdf)
[Localdev Foundation Balance Sheet 2022-05-01-20.pdf](https://github.com/drud/ddev/files/8655186/Localdev.Foundation.Balance.Sheet.2022-05-01-20.pdf)

**Comment by rfay** [comment]
Proposed Test Runner Purchase Plan:

Mac mini amd64: 2 $500-$650 each $1400

Mac mini arm64: 1 $650 each. $700

Windows MIni-PC 8GB/256GB with Windows 10/11 Pro, About $500. Purchase only one until usability verified, then purchase 4 more.  Something like [beelink](https://smile.amazon.com/Beelink-Processor-Windows-Computer-Business/dp/B09H5961YN?ref_=ast_sto_dp&th=1).  $400-$2000

Total investment for 2022: $4100

**Comment by chrisfromredfin** [comment]
Raw notes: https://docs.google.com/document/d/1WZxyEX45NAmirvScqFMdlRQKowKu0naHt39ybJJLvOk/edit#

**Reply by andy-blum** [reply]
Thanks! converted to .md and inserted!

**Comment by rfay** [comment]
Added the recording to the top comment
